api/src/main/java/org/pmiops/workbench/profile/ProfileService.java

```diff
@@ -334,7 +344,22 @@ private void validateStringLength(String field, String fieldName, int max, int m
     }
   }
 
-  public List<Profile> listAllProfiles() {
-    return userService.getAllUsers().stream().map(this::getProfile).collect(Collectors.toList());
+  public void adminUpdateProfile(Profile updatedProfile) {
+    final DbUser user = profileMapper.profileToDbUser(updatedProfile);
+
+    // TODO RW-5104: move these DB null fixes to mapper or other common code
+
+    user.getPageVisits().forEach(dbPageVisit -> dbPageVisit.setUser(user));
```

Comment:
I don't understand what's going on with this. How did the user get page visits that don't know she is the user?